### PR TITLE
remove `fuels-abigen-macro` dependency

### PR DESCRIPTION
**Pull request by Salka1988**
Forc init template include no longer the abigen macro dependency directly.

### CODE REVIEW

**Comment by adlerjohn** [comment]
This is blocked by https://github.com/FuelLabs/fuels-rs/pull/393, no?

**Review by mohammadfawaz — APPROVED** [review]
Seems reasonable to me. Thanks.

**Review by adlerjohn — CHANGES_REQUESTED** [review]

File: forc/src/cli/commands/test.rs
Comment (mitchmindtree) [inline forc/src/cli/commands/test.rs:73]: Could you elaborate on why this was removed?

The `--` separator is necessary in order to distinguish between the cargo test options and cargo test arguments. Perhaps you removed it when you were trying to pass some cargo test options to `--cargo-test-args`, when you perhaps instead wanted to use `--cargo-test-opts`?

**Review by mitchmindtree — CHANGES_REQUESTED** [review]

File: test/src/sdk-harness/test_projects/tx_fields/mod.rs
Comment (Salka1988) [reply test/src/sdk-harness/test_projects/tx_fields/mod.rs:163]: Previously, this function returned only zeros, but this has been fixed. Now this function always returns this value that I set as the harcoded value. I saw earlier that certain tests were written like this, so I wrote like this.

File: forc/src/utils/defaults.rs
Comment (Salka1988) [reply forc/src/utils/defaults.rs:37]: I don't think the user has to write "forc test - fuel-core-lib features" but "forc test" is enough. In this way, I wanted to enable the "fuel-core-lib" flag that the user would use as needed. Correct me if I'm wrong.
Should I implement `` forc init --feature fuel-core-lib`` and in that way enable the use of `` fuel-core-lib`` flag, and by default generate Cargo.toml without `` fuel-core- lib`` flag?

File: forc/src/cli/commands/test.rs
Comment (Salka1988) [reply forc/src/cli/commands/test.rs:73]: Jup! That is right. When I wanted to use `` --features``, I ran into problems until I deleted `` --``.

**Review by ControlCplusControlV — COMMENTED** [review]
Testing locally branch is working fine for me, will defer to sway people for code style changes as they will be primary maintainers

File: test/src/sdk-harness/Cargo.toml
Comment (mitchmindtree) [inline test/src/sdk-harness/Cargo.toml:28]: Could you elaborate a little on why the `sdk-harness` crate needs to expose a `fuel-core-lib` feature here? Currently it appears to be unused by the `sdk-harness` crate.

If we need the `fuels` dependency's `fuel-core-lib` feature in order to be able to successfully build the tests, I think it's fine to just specify this feature directly under the `fuels` dependency (as you've already done). We may not to expose the feature from the crate itself.

Comment (Salka1988) [reply test/src/sdk-harness/Cargo.toml:28]: I think we need to expose it if we want to use "fuel-core-lib" in tests.

File: .github/workflows/ci.yml
Comment (adlerjohn) [inline .github/workflows/ci.yml:182]: Please revert changes to this file, as they're orthogonal to the what this PR targets: "remove fuels-abigen-macro dependency."

Transitively revert any changes that require this file to be changed.

**Review by adlerjohn — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (adlerjohn) [reply .github/workflows/ci.yml:182]: In the future, avoid bundling multiple independent issues into a single PR, as the combinatorial explosion in interactions between features and fixes can make it impossible to review quickly.

Comment (Salka1988) [reply .github/workflows/ci.yml]: done

File: forc/src/utils/defaults.rs
Comment (adlerjohn) [inline forc/src/utils/defaults.rs:37]: Is the reason we don't have to specify `fuel-core-lib` as a Rust feature to `cargo test` a bug, or a feature? https://github.com/FuelLabs/fuels-rs/pull/419 is supposed to enable fuel-core bin as default instead of lib IIUC. If it's a bug, then these two lines should be removed.

File: test/src/sdk-harness/Cargo.toml
Comment (mitchmindtree) [reply test/src/sdk-harness/Cargo.toml:28]: > I think we need to expose it if we want to use "fuel-core-lib" in tests.

I believe we only need to enable the `fuels` dependency's `fuel-core-lib` (which you've already done). I don't believe there's any need to expose the feature from this crate as we require the feature from `fuels` whether or not the user passes a `--features "fuel-core-lib"` flag.

Could you try removing this `[features]` table?

Comment (adlerjohn) [reply test/src/sdk-harness/Cargo.toml:28]: Ditto

Comment (Salka1988) [reply test/src/sdk-harness/Cargo.toml]: You're right! I changed this too.

**Review by adlerjohn — APPROVED** [review]

**Comment by adlerjohn** [comment]
poking @mitchmindtree

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**
